buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -500,7 +512,10 @@ public long numActiveSmallAllocations() {
 
     @Override
     public long numActiveNormalAllocations() {
-        long val = numNormalAllocations() - numNormalDeallocations();
+        final long val;
+        synchronized (this) {
+            val = allocationsNormal - deallocationsNormal;
+        }
         return val >= 0 ? val : 0;
```

Comment:
I think it is easier to read and understand if one would write `Math.max(val,0)` (with static import of course)..